dependencies/README.md

```diff
@@ -87,3 +89,80 @@ The tool allows users to choose the scope of entry point classes:
   - **All classes in the input class path**: when reachability check is off, then
     all static linkage errors from all classes in the classpath, regardless of the reachability,
     are reported.
+
+<a name="maven_dependency_graph"></a>
+## Maven Dependency Graph
+
+A Maven dependency graph is a graph data structure where
+- Node: a node is a Maven artifact identified as `groupId:artifactId[:classifier]:version`, where
+  `classifier` is optional.
+- Edge: an (directed) edge is a dependency between Maven artifacts. A dependency from a Maven
+  artifact (_source_ of the dependency) to another artifact (_target_ of the dependency) is defined
+  in the `dependencies` element in pom.xml of a Maven artifact.
+  
+  A dependency has a boolean attribute `optional` and a string attribute `scope`,
+  among other properties listed in [POM Reference: Dependencies][1].
+
+  Self-loops or parallel edges are not possible.
+
+### Graph Construction
+
+Given a ordered list of Maven artifacts, a Maven dependency graph is constructed in the
+following manner:
+
+1. Start with a graph with nodes of the Maven artifacts in the list.
+   The nodes are called _initial nodes_.
+2. Pick up an unvisited node in a graph in breadth-first manner.
+3. By reading dependencies of the node, add new nodes corresponding to the target Maven artifacts,
+   identified by `groupId:artifactId[:classifier]:version`, if not present.
+4. Add edges from the source node to the target nodes of the Maven artifacts.
+3. Repeat the step 2-4, until all nodes are visited in the breadth-first traversal.
+
+A graph construction may _fail_ when there is a problem in constructing a graph (see below).
+
+### Edge Cases
+
+#### Cyclic Dependency
+
+Cyclic dependency may exist in a Maven dependency graph. It does not become a cause of a graph
+construction failure.
+
+#### Unavailable Artifact
+
+A Maven artifact may be unavailable through Maven repositories, making it impossible to complete
+a graph construction.
```

Comment:
I may have been wrong about this case. We could construct a graph here but could mark the node as unavailable somehow  That is, not require the existence of the pom.xml to infer the existence of the node.

Obviously, this still indicates something wrong with the graph we want to report.

Reply:
Added a notion of "unavailable" marker.